app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -109,56 +119,28 @@
       _.each(ctrl.envFromEntries, function(entry) {
         if(entry) {
           if(entry.configMapRef && !canI('configmaps', 'get')) {
-            entry.isReadonlyValue = true;
+            entry.isReadonly = true;
           }
 
           if(entry.secretRef && !canI('secrets', 'get')) {
-            entry.isReadonlyValue = true;
+            entry.isReadonly = true;
           }
         }
       });
-    };
-
-    var getReferenceValue = function(option) {
-      var referenceValue;
 
-      switch(option.kind) {
-        case 'ConfigMap':
-          referenceValue = _.find(ctrl.envFromEntries, {configMapRef: {name: option.metadata.name}});
-          break;
-        case 'Secret':
-          referenceValue = _.find(ctrl.envFromEntries, {secretRef: {name: option.metadata.name}});
-          break;
-      }
+      _.each(ctrl.envFromSelectorOptions, function(option) {
+        _.each(getReferenceValues(option), function(val, i) {
```

Comment:
You should create two maps of options `secretsByName` and `configMapsByName` when options change. Then you can index directly into the map using `configMapsByName[option.configMapRef.name]`, etc.